include/xsf/specfun/specfun.h

```diff
@@ -892,10 +892,11 @@ inline double chgu(double x, double a, double b, int *md, int *isfer) {
         }
     } else {
         if (b <= a) {
+            b00 = b;
             a -= b - 1.0;
             b = 2.0 - b;
             hu = chguit(x, a, b, &id);
-            hu *= pow(x, 1.0 - b);
+            hu *= pow(x, 1.0 - b00);
             *md = 4;
         } else if (bn && (~il1)) {
```

Comment:
Here we want to use the unmodified `b`.

```for
           IF (B.LE.A) THEN
              A00=A
              B00=B 		! ........... save B, B00 <- B
              A=A-B+1.0D0
              B=2.0D0-B 	! ........... modify B, B <- 2 - B
              CALL CHGUIT(A,B,X,HU,ID)
              HU=X**(1.0D0-B00)*HU 		! use unmodified B00
              A=A00
              B=B00
              MD=4
           ELSE IF (BN.AND.(.NOT.IL1)) THEN
              CALL CHGUBI(A,B,X,HU,ID)
              MD=3
           ENDIF
```

xref: https://github.com/scipy/scipy/blob/9d08eb58205fb669ff72c60080b18be61b14d8e1/scipy/special/specfun/specfun.f#L4705-L4715